.github/workflows/ci.yml

```diff
@@ -0,0 +1,191 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:   # run for all PRs (parity with GitLab MR rule)
+    types: [opened, synchronize, reopened, ready_for_review]
```

Comment:
Purge this out--seems unnecessary. In general, this CI config file has far too much duplication/complexity, and merge conflicts need to be resolved against the more concise CI config I added in gh-9.